src/lib.rs

```diff
@@ -225,6 +233,46 @@ impl KeyPair {
         let pk = sk.public_key()?;
         Ok(KeyPair { sk, pk })
     }
+
+    // Generate a strong prime safe key pair used for Partial Blinding RSA
+    pub fn generate_strong_pair(modulus_bits: usize,) -> Result<KeyPair, Error> {
+        if modulus_bits % 2 != 0 {
+            return Err(Error::UnsupportedParameters); 
+        }
+
+        let p = Self::safe_prime(modulus_bits / 2);
+        let mut q = Self::safe_prime(modulus_bits / 2);
+        let mut n = &p * & q;
+
+        while p == q || n.bits() != modulus_bits {
+            q = Self::safe_prime(modulus_bits / 2);
+            n = &p * &q;
+        }
+
+        let phi = (&p - BigUint::one()) * (&q - BigUint::one());
+        let e = BigUint::from(65537u32);
+        let d = e.clone().mod_inverse(phi.clone()).unwrap().to_biguint().unwrap(); 
+
+        let sk = RsaPrivateKey::from_components(n.clone(), e, d, vec![p, q]).map_err(|_| Error::InvalidKey)?; 
```

Comment:
n isn't used after this point so no need to clone it
```suggestion
        let sk = RsaPrivateKey::from_components(n, e, d, vec![p, q]).map_err(|_| Error::InvalidKey)?; 
```